pkg/golinters/gci/gci.go

```diff
@@ -1,61 +1,118 @@
 package gci
 
 import (
+	"bytes"
 	"fmt"
+	"io"
+	"os"
 	"strings"
 
+	gcicfg "github.com/daixiang0/gci/pkg/config"
+	"github.com/daixiang0/gci/pkg/gci"
+	"github.com/daixiang0/gci/pkg/log"
+	"github.com/shazow/go-diff/difflib"
 	"golang.org/x/tools/go/analysis"
 
 	"github.com/golangci/golangci-lint/pkg/config"
 	"github.com/golangci/golangci-lint/pkg/goanalysis"
-	"github.com/golangci/golangci-lint/pkg/golinters/gci/internal"
+	"github.com/golangci/golangci-lint/pkg/golinters/internal"
+	"github.com/golangci/golangci-lint/pkg/lint/linter"
 )
 
 const linterName = "gci"
 
-const prefixSeparator = "¤"
+type differ interface {
+	Diff(out io.Writer, a io.ReadSeeker, b io.ReadSeeker) error
+}
 
 func New(settings *config.GciSettings) *goanalysis.Linter {
-	a := internal.NewAnalyzer()
-
-	var cfg map[string]map[string]any
-	if settings != nil {
-		var sections []string
-		for _, section := range settings.Sections {
-			if strings.HasPrefix(section, "prefix(") {
-				sections = append(sections, strings.ReplaceAll(section, ",", prefixSeparator))
-				continue
+	log.InitLogger()
+	_ = log.L().Sync()
+
+	diff := difflib.New()
+
+	a := &analysis.Analyzer{
+		Name: linterName,
+		Doc:  goanalysis.TheOnlyanalyzerDoc,
+		Run:  goanalysis.DummyRun,
+	}
+
+	return goanalysis.NewLinter(
+		linterName,
+		a.Doc,
+		[]*analysis.Analyzer{a},
+		nil,
+	).WithContextSetter(func(lintCtx *linter.Context) {
+		a.Run = func(pass *analysis.Pass) (any, error) {
+			err := run(lintCtx, pass, settings, diff)
```

Comment:
I think you're not a fan of it, but `run` either returns an error or `nil` and we don't wrap it so I guess we could always just return the `run` result. Or leave it if you like the explicit error handling 👍 


```suggestion
			return nil, run(lintCtx, pass, settings, diff)
```

Reply:
Yes I'm a fan of oneliner :smile_cat: 

In all cases, I rewritten all the formatters, you will see that quickly.